Updated a few scripts to use new arpa2fst options

kkm000 opened the pull request:
I converted a few scripts by hand: LibriSpeech because they are easy to test here (everything is downloaded and set up), WSJ because they are the WSJ, and AMI because they start with the 'A', for the lack of a better explanation. Of these, I ran LibriSpeech to convert and confirmed `utils/validate_lang.pl` succeed. Others are untested (most I cannot test anyway).

There are 39 more occurrences, but they formatted differently so scripting the changes is not easy.(`find -name '*.sh' | sort | xargs grep --color -nP 'arpa2fst -(?!-)'` to find).

Is the goal to convert them all? I can do them by hand a few at a time when I get bored of everything else.

kkm000 commented:
@danpovey: so do you think it would be a good idea to change all scripts like that?

danpovey commented:
Yes I do think it would be a good idea to change them all, but I'd rather
do a few, test them, wait for a problems, and then do the others.
Dan


On Fri, Apr 1, 2016 at 12:51 AM, Kirill Katsnelson <notifications@github.com
> wrote:

> @danpovey <https://github.com/danpovey>: so do you think it would be a
> good idea to change all scripts like that?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/634#issuecomment-204247996>
>

kkm000 commented:
I've tested only LibriSpeech. I think the PR(s) should be just eyeballed carefully, mostly. The change is regular enough.

danpovey commented on egs/ami/s5/local/ami_format_data.sh:
```diff
@@ -15,25 +15,12 @@ arpa_lm=$1
-     --osymbols=data/lang_test/words.txt  --keep_isymbols=false --keep_osymbols=false | \
-    fstrmepsilon | fstarcsort --sort_type=ilabel > data/lang_test/G.fst
-  fstisstochastic data/lang_test/G.fst
+  arpa2fst --disambig-symbol=#0 \
```
instead of - | data/lang_test/G.fst you could just write data/lang_test/G.fst.

kkm000 replied:
Rather `... | arpa2fst - data/lang_test/G.fst` -- is that what you mean? I can do that as well, just a matter of style. It looks to me most pipelines in the scripts end in a redirection. But I am totally ambivalent about these, which way would you prefer?

danpovey replied:
I just don't like the look of having the explicit "-" followed by a
redirection- it seems redundant.  I.e. if you're going write - and then
write the filename, you could just write the filename.  I'd put the
filename there directly.

On Fri, Apr 1, 2016 at 1:39 AM, Kirill Katsnelson <notifications@github.com>
wrote:

> In egs/ami/s5/local/ami_format_data.sh
> <https://github.com/kaldi-asr/kaldi/pull/634#discussion_r58164126>:
>
> >  gunzip -c "$arpa_lm" | \
> > -   grep -v '<s> <s>' | \
> > -   grep -v '</s> <s>' | \
> > -   grep -v '</s> </s>' | \
> > -   arpa2fst - | fstprint | \
> > -   utils/remove_oovs.pl /dev/null | \
> > -   utils/eps2disambig.pl | utils/s2eps.pl | fstcompile --isymbols=data/lang_test/words.txt \
> > -     --osymbols=data/lang_test/words.txt  --keep_isymbols=false --keep_osymbols=false | \
> > -    fstrmepsilon | fstarcsort --sort_type=ilabel > data/lang_test/G.fst
> > -  fstisstochastic data/lang_test/G.fst
> > +  arpa2fst --disambig-symbol=#0 \
>
> Rather ... | arpa2fst - data/lang_test/G.fst -- is that what you mean? I
> can do that as well, just a matter of style. It looks to me most pipelines
> in the scripts end in a redirection. But I am totally ambivalent about
> these, which way would you prefer?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/634/files/7f89580915a7568b2fa897570b041cfcb2e96108#r58164126>
>

keli78 replied:
Hi Dan,
I'll test this pull request on wsj today.
Ke

On Fri, Apr 1, 2016 at 1:43 AM, Daniel Povey <notifications@github.com>
wrote:

> In egs/ami/s5/local/ami_format_data.sh
> <https://github.com/kaldi-asr/kaldi/pull/634#discussion_r58164308>:
>
> >  gunzip -c "$arpa_lm" | \
> > -   grep -v '<s> <s>' | \
> > -   grep -v '</s> <s>' | \
> > -   grep -v '</s> </s>' | \
> > -   arpa2fst - | fstprint | \
> > -   utils/remove_oovs.pl /dev/null | \
> > -   utils/eps2disambig.pl | utils/s2eps.pl | fstcompile --isymbols=data/lang_test/words.txt \
> > -     --osymbols=data/lang_test/words.txt  --keep_isymbols=false --keep_osymbols=false | \
> > -    fstrmepsilon | fstarcsort --sort_type=ilabel > data/lang_test/G.fst
> > -  fstisstochastic data/lang_test/G.fst
> > +  arpa2fst --disambig-symbol=#0 \
>
> I just don't like the look of having the explicit "-" followed by a
> redirection- it seems redundant. I.e. if you're going write - and then
> write the filename, you could just write the filename. I'd put the filename
> there directly.
> … <#m_5623312248150552701_>
> On Fri, Apr 1, 2016 at 1:39 AM, Kirill Katsnelson <
> notifications@github.com> wrote: In egs/ami/s5/local/ami_format_data.sh <#634
> (comment)
> <https://github.com/kaldi-asr/kaldi/pull/634#discussion_r58164126>>: >
> gunzip -c "$arpa_lm" | \ > - grep -v '<s> <s>' | \ > - grep -v '</s> <s>' |
> \ > - grep -v '</s> </s>' | \ > - arpa2fst - | fstprint | \ > - utils/
> remove_oovs.pl /dev/null | \ > - utils/eps2disambig.pl | utils/s2eps.pl |
> fstcompile --isymbols=data/lang_test/words.txt \ > -
> --osymbols=data/lang_test/words.txt --keep_isymbols=false
> --keep_osymbols=false | \ > - fstrmepsilon | fstarcsort --sort_type=ilabel
> > data/lang_test/G.fst > - fstisstochastic data/lang_test/G.fst > +
> arpa2fst --disambig-symbol=#0 \ Rather ... | arpa2fst -
> data/lang_test/G.fst -- is that what you mean? I can do that as well, just
> a matter of style. It looks to me most pipelines in the scripts end in a
> redirection. But I am totally ambivalent about these, which way would you
> prefer? — You are receiving this because you were mentioned. Reply to this
> email directly or view it on GitHub <
> https://github.com/kaldi-asr/kaldi/pull/634/files/7f89580915a7568b2fa897570b041cfcb2e96108#r58164126>
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/634/files/7f89580915a7568b2fa897570b041cfcb2e96108#r58164308>
>



-- 
Ke Li
Dept. of Electrical and Computer Engineering
Johns Hopkins University
Email: kli26@jhu.edu

kkm000 replied:
I'll remove the redirects.

kkm000 replied:
>I'll remove the redirects.

Done in https://github.com/kaldi-asr/kaldi/pull/634/commits/c7bc7585

danpovey commented on egs/wsj/s5/local/wsj_format_data.sh:
```diff
@@ -49,22 +49,8 @@ for lm_suffix in bg tgpr tg bg_5k tgpr_5k tg_5k; do
-    utils/eps2disambig.pl | utils/s2eps.pl | fstcompile --isymbols=$test/words.txt \
-      --osymbols=$test/words.txt  --keep_isymbols=false --keep_osymbols=false | \
-     fstrmepsilon | fstarcsort --sort_type=ilabel > $test/G.fst
+    arpa2fst --disambig-symbol=#0 \
```
does your arpa2fst now get rid of the strange things like <s> <s> and </s> <s> by itself, or somehow ignore them?  I think these exist in the LMs that come with WSJ, and they can cause problems.

kkm000 replied:
Yes it does. <s> can only be the first word of an n-gram, and </s> the last.

danpovey commented:
@keli78, actually please run this far enough that it does 'big-dict' decoding (with one of the _bd_ lang directories).  I want to make sure all that stuff is working.

danpovey commented:
Thanks.  I'm still waiting for @keli78 to test this on WSJ before merging.

danpovey commented:
I'm looking at the output where @keli78  was testing it...
there are situations where we get a ton of these warnings:
WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:306) In line 797081: Skipping n-gram with invalid BOS/EOS placement
WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:306) In line 797082: Skipping n-gram with invalid BOS/EOS placement
WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:306) In line 797083: Skipping n-gram with invalid BOS/EOS placement
WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:306) In line 797084: Skipping n-gram with invalid BOS/EOS placement
WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:306) In line 797085: Skipping n-gram with invalid BOS/EOS placement
WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:306) In line 797086: Skipping n-gram with invalid BOS/EOS placement

I think some kind of mechanism to limit how many can be printed per program would be desirable.  (This is with old LMs that are distributed with WSJ).


There is another problem- and this is likely independent of this PR, it's an issue with the ARPA reading code and how it interacts with the old LMs distributed by Lincoln Labs.
We're getting the error
\\data\\ section missing or empty.
This seems to be because one of the LMs distributed with WSJ has a bunch of crap in the header before the \\data\\ marker:
```
***************************************************************************
   Copyright (C) 1990, 1991 MIT Lincoln Laboratory.  All rights reseved.
***************************************************************************
./perplex -c -u -Vwlist20o.nvp -gbg-train64.nvp -tbg-dev-test20.nvp -o
Mapping unknown words to <UNK>
compact mode 3
minimum word count=55604
biclean n2gram: 1466230 -> 1460564
nwtoken=38500240, nw=19982, 19982 1-grams, 1460564 2-grams
bsep.cnt=1618226, esep.cnt=1612251, unk.cnt=719777
Back-off prob
bigram backoff fix=34

Bigram back-off language model.
log10 probs and log10 back-off weights (bo_wt)


p(wd2|wd1)= if(bigram exists) p_2(wd1,wd2)
            else              bo_wt_1(wd1)*p_1(wd2)

data formats:
beginning of data mark: \data\
[wd_code]               # if word codes are used
ngram 1=nr              # number of unigrams
[ngram 2=nr]            # number of bigrams
[ngram 3=nr]            # number of trigrams

unigram format:
\1-grams:
p_1     wd      bo_wt_1 [wd_code]
(if wd_codes are given, they will be used in the bigrams and trigrams)
(wd_codes are base 64 numbers: 0-9A-Za-z{})

bigram format:
\2-grams:
p_2     wd1 wd2


end of data mark: \end\

\data\
ngram 1=19982
ngram 2=1460564

\1-grams:
-1.7282 <UNK>   -0.7021
```
So you see, you have to read until you see a line with *just* \data\ on it.  The current code seems to break as soon as it finds a line that starts with backslash, thanks to the 'break' statement on arpa-file-parser.cc:94.


@keli78, after Kirill fixes this issue, please do git pull, and make sure you recompile your code ("make" in src/) before testing this again.  I notice you are getting the error `--> ERROR: data/lang_nosp_test_tgpr/G.fst is not ilabel sorted`, which seems to be because you have not compiled recently enough-- the program does that now.

kkm000 commented:
Actually, neither of these issues is with the current PR. The new arpa2fst does same things in both the "compatible" (good old `arpa2fst - -`) and "modern" modes. I though about limiting warnings but never actually implemented that. And crap before \data\ in that file is a royal crap indeed.

 I'll make a fix quickly but it should be a separate PR.

keli78 commented:
Hi Dan,
You're right. This error :  --> ERROR: data/lang_nosp_test_tgpr/G.fst is
not ilabel sorted has been fixed now. But I still got the same error as
before:
arpa2fst --disambig-symbol=#0
--read-symbol-table=data/lang_nosp_test_bg_5k/words.txt - data/lang_nosp_te
    st_bg_5k/G.fst

ERROR (arpa2fst:Read():arpa-file-parser.cc:122) in line 29: \data\ section
missing or empty.

ERROR (arpa2fst:Read():arpa-file-parser.cc:122) in line 29: \data\ section
missing or empty.

I'll rerun it after the problem above is fixed.

Ke

On Sun, Apr 3, 2016 at 12:13 AM, Kirill Katsnelson <notifications@github.com
> wrote:

> Actually, neither of these issues is with the current PR. The new arpa2fst
> does same things in both the "compatible" (good old arpa2fst - -) and
> "modern" modes. I though about limiting warnings but never actually
> implemented that. And crap before \data\ in that file is a royal crap
> indeed.
>
> I'll make a fix quickly but it should be a separate PR.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/634#issuecomment-204864389>
>



-- 
Ke Li
Dept. of Electrical and Computer Engineering
Johns Hopkins University
Email: kli26@jhu.edu

kkm000 commented:
@keli78: I just rebased this change on the committed fix, you can rerun the script now.

@danpovey: 
>Actually, neither of these issues is with the current PR.
-- I was wrong saying that, the flood of warnings is not because of arpa2fst per se, but rather because I removed the lines that grep out some invalid BOS/EOS patterns. I'm working on a fix.

kkm000 commented:
Setting to WIP until I addresss the [warnings](https://github.com/kaldi-asr/kaldi/pull/634#issuecomment-204861154).

kkm000 commented:
Blocked by #649

danpovey commented:
@keli78 , you should be able to re-test now-- you could do
`git pull upstream/master`
(assuming this repo is named 'upstream' in your repo)
to get the latest changes to re-test with.

keli78 commented:
Sure, I'm testing it now.

On Mon, Apr 4, 2016 at 8:17 PM, Daniel Povey <notifications@github.com>
wrote:

> @keli78 <https://github.com/keli78> , you should be able to re-test now--
> you could do
> git pull upstream/master
> (assuming this repo is named 'upstream' in your repo)
> to get the latest changes to re-test with.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/634#issuecomment-205555275>
>



-- 
Ke Li
Dept. of Electrical and Computer Engineering
Johns Hopkins University
Email: kli26@jhu.edu

danpovey commented:
Kirill, from what I hear from @kili78, it looks like this is going smoothly.  
I'll merge this now.  We can wait a little for problems, then do the other ones- don't forget!

keli78 commented:
Hi Dan,
I'm on the decoding stage of nnet2. By far there is no error, but still
many warning lines (maybe this doesn't matter much?) The log file is here:
/export/a16/keli1/kaldi-arpa2fst-check/egs/wsj/s5/log4.wsj
Do you want me to stop it?
Ke

On Tue, Apr 5, 2016 at 6:32 PM, Daniel Povey <notifications@github.com>
wrote:

> Kirill, from what I hear from @kili78, it looks like this is going
> smoothly.
>
> I'll merge this now. We can wait a little for problems, then do the other
> ones- don't forget!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/634#issuecomment-206015043>
>



-- 
Ke Li
Dept. of Electrical and Computer Engineering
Johns Hopkins University
Email: kli26@jhu.edu

danpovey commented:
OK, you can stop it; just check the results are in the right ballpark.
kirill: we seem to get an excessive number of these warnings, they don't
seem rate limited:

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787245
[-1.5986  <s> <s> <UNK>] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787246
[-1.6214  <s> <s> A] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787247
[-2.878   <s> <s> A.] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787248
[-4.7058  <s> <s> AARON] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787249
[-5.4328  <s> <s> ABANDONED] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787250
[-5.7249  <s> <s> ABANDONING] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787251
[-4.8686  <s> <s> ABBEY] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787252
[-4.6925  <s> <s> ABBOTT] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787253
[-5.6023  <s> <s> ABBOUD] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787254
[-5.0649  <s> <s> ABBY] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787255
[-5.6023  <s> <s> ABDUL] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787256
[-5.6023  <s> <s> ABE] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787257
[-4.5482  <s> <s> ABITIBI] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787258
[-5.8942  <s> <s> ABLE] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787259
[-5.8942  <s> <s> ABOARD] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787260
[-5.8942  <s> <s> ABOLISHING] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787261
[-4.8686  <s> <s> ABORTION] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787262
[-2.8174  <s> <s> ABOUT] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787263
[-4.1355  <s> <s> ABOVE] skipped: n-gram has invalid BOS/EOS placement

WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787264
[-4.7486  <s> <s> ABRAHAM] skipped: n-gram has invalid BOS/EOS placement


(but the ones like this

WARNING (arpa2fst:Read():arpa-file-parser.cc:192) line 483019 [-4.8745  OF
CONISTON'S] skipped: word 'CONISTON'S' not in symbol table

do seem to have a max. number).



On Tue, Apr 5, 2016 at 8:24 PM, Ke Li <notifications@github.com> wrote:

> Hi Dan,
> I'm on the decoding stage of nnet2. By far there is no error, but still
> many warning lines (maybe this doesn't matter much?) The log file is here:
> /export/a16/keli1/kaldi-arpa2fst-check/egs/wsj/s5/log4.wsj
> Do you want me to stop it?
> Ke
>
> On Tue, Apr 5, 2016 at 6:32 PM, Daniel Povey <notifications@github.com>
> wrote:
>
> > Kirill, from what I hear from @kili78, it looks like this is going
> > smoothly.
> >
> > I'll merge this now. We can wait a little for problems, then do the other
> > ones- don't forget!
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly or view it on GitHub
> > <https://github.com/kaldi-asr/kaldi/pull/634#issuecomment-206015043>
> >
>
>
>
> --
> Ke Li
> Dept. of Electrical and Computer Engineering
> Johns Hopkins University
> Email: kli26@jhu.edu
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/634#issuecomment-206045461>
>

kkm000 commented:
The default limit is 30 for all warnings _in n-gram record processing_. The warning about 0 n-grams in an order (remember, the one you tweaked couple days ago) is therefore exempted from counting, but that is currently the only exception. Could it be that the counts you are seeing are still under the limit? I count 20 in your sample.

cf [this](https://github.com/kaldi-asr/kaldi/blob/master/src/lm/arpa-file-parser.cc#L191)
```c++
                  if (ShouldWarn())
                    KALDI_WARN << LineReference() << " skipped: word '"
                               << col[1 + index] << "' not in symbol table";
```
and [this](https://github.com/kaldi-asr/kaldi/blob/master/src/lm/arpa-lm-compiler.cc#L186)
```c++
    if (parent_->ShouldWarn())
      KALDI_WARN << parent_->LineReference()
                 << " skipped: no parent (n-1)-gram exists";
```
But I'll run and check anyway. And I'll do the other scripts, sure.

kkm000 commented:
@danpovey: Aw shucks, there's really a bug, you are talking of a different message! I'll fix now.

keli78 commented:
I just stopped it. The results I checked (on tri1 tri2a tri3b and tri4b
...) are approximately the same as the baselines.
Ke

On Tue, Apr 5, 2016 at 9:05 PM, Daniel Povey <notifications@github.com>
wrote:

> OK, you can stop it; just check the results are in the right ballpark.
> kirill: we seem to get an excessive number of these warnings, they don't
> seem rate limited:
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787245
> [-1.5986 <s> <s> <UNK>] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787246
> [-1.6214 <s> <s> A] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787247
> [-2.878 <s> <s> A.] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787248
> [-4.7058 <s> <s> AARON] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787249
> [-5.4328 <s> <s> ABANDONED] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787250
> [-5.7249 <s> <s> ABANDONING] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787251
> [-4.8686 <s> <s> ABBEY] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787252
> [-4.6925 <s> <s> ABBOTT] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787253
> [-5.6023 <s> <s> ABBOUD] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787254
> [-5.0649 <s> <s> ABBY] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787255
> [-5.6023 <s> <s> ABDUL] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787256
> [-5.6023 <s> <s> ABE] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787257
> [-4.5482 <s> <s> ABITIBI] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787258
> [-5.8942 <s> <s> ABLE] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787259
> [-5.8942 <s> <s> ABOARD] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787260
> [-5.8942 <s> <s> ABOLISHING] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787261
> [-4.8686 <s> <s> ABORTION] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787262
> [-2.8174 <s> <s> ABOUT] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787263
> [-4.1355 <s> <s> ABOVE] skipped: n-gram has invalid BOS/EOS placement
>
> WARNING (arpa2fst:ConsumeNGram():arpa-lm-compiler.cc:307) line 787264
> [-4.7486 <s> <s> ABRAHAM] skipped: n-gram has invalid BOS/EOS placement
>
>
> (but the ones like this
>
> WARNING (arpa2fst:Read():arpa-file-parser.cc:192) line 483019 [-4.8745 OF
> CONISTON'S] skipped: word 'CONISTON'S' not in symbol table
>
> do seem to have a max. number).
>
>
>
> On Tue, Apr 5, 2016 at 8:24 PM, Ke Li <notifications@github.com> wrote:
>
> > Hi Dan,
> > I'm on the decoding stage of nnet2. By far there is no error, but still
> > many warning lines (maybe this doesn't matter much?) The log file is
> here:
> > /export/a16/keli1/kaldi-arpa2fst-check/egs/wsj/s5/log4.wsj
> > Do you want me to stop it?
> > Ke
> >
> > On Tue, Apr 5, 2016 at 6:32 PM, Daniel Povey <notifications@github.com>
> > wrote:
> >
> > > Kirill, from what I hear from @kili78, it looks like this is going
> > > smoothly.
> > >
> > > I'll merge this now. We can wait a little for problems, then do the
> other
> > > ones- don't forget!
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/kaldi-asr/kaldi/pull/634#issuecomment-206015043>
> > >
> >
> >
> >
> > --
> > Ke Li
> > Dept. of Electrical and Computer Engineering
> > Johns Hopkins University
> > Email: kli26@jhu.edu
> >
> > —
> > You are receiving this because you modified the open/close state.
> > Reply to this email directly or view it on GitHub
> > <https://github.com/kaldi-asr/kaldi/pull/634#issuecomment-206045461>
>
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/634#issuecomment-206058995>
>



-- 
Ke Li
Dept. of Electrical and Computer Engineering
Johns Hopkins University
Email: kli26@jhu.edu
